From: Manaf Meethalavalappu Pallikunhi <quic_manafm@quicinc.com>
Date: Wed, 25 Mar 2026 16:50:01 +0530
Subject: [PATCH] arm64: dts: qcom: sm8750: Add thermal zones for TSENS

Define 47 thermal zones covering CPU, GPU, NPU, modem, camera,
video and DDR sensors. Hot trip at 120C, critical at 125C.

Signed-off-by: Gaurav Kohli <gaurav.kohli@oss.qualcomm.com>
[ROCKNIX: backported from upstream commit 0a78d270a40ecec6bae69920bb4319442b98cf24]
---
diff --git a/arch/arm64/boot/dts/qcom/sm8750.dtsi b/arch/arm64/boot/dts/qcom/sm8750.dtsi
--- a/arch/arm64/boot/dts/qcom/sm8750.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm8750.dtsi
@@ -6274,3 +6274,335 @@
 		};
 	};
-};
+
+	thermal-zones {
+		aoss0-thermal {
+			thermal-sensors = <&tsens0 0>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				aoss0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-0-0-thermal {
+			thermal-sensors = <&tsens0 1>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-0-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-0-1-thermal {
+			thermal-sensors = <&tsens0 2>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-0-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-1-0-thermal {
+			thermal-sensors = <&tsens0 3>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-1-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-1-1-thermal {
+			thermal-sensors = <&tsens0 4>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-1-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-2-0-thermal {
+			thermal-sensors = <&tsens0 5>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-2-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-2-1-thermal {
+			thermal-sensors = <&tsens0 6>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-2-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-3-0-thermal {
+			thermal-sensors = <&tsens0 7>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-3-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-3-1-thermal {
+			thermal-sensors = <&tsens0 8>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-3-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-4-0-thermal {
+			thermal-sensors = <&tsens0 9>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-4-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-4-1-thermal {
+			thermal-sensors = <&tsens0 10>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-4-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-5-0-thermal {
+			thermal-sensors = <&tsens0 11>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-5-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-0-5-1-thermal {
+			thermal-sensors = <&tsens0 12>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-0-5-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpuss_0_0_thermal: cpuss-0-0-thermal {
+			thermal-sensors = <&tsens0 13>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpuss-0-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpuss_0_1_thermal: cpuss-0-1-thermal {
+			thermal-sensors = <&tsens0 14>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpuss-0-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		aoss1-thermal {
+			thermal-sensors = <&tsens1 0>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				aoss1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-1-0-0-thermal {
+			thermal-sensors = <&tsens1 1>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-1-0-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-1-0-1-thermal {
+			thermal-sensors = <&tsens1 2>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-1-0-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-1-1-0-thermal {
+			thermal-sensors = <&tsens1 3>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-1-1-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpu-1-1-1-thermal {
+			thermal-sensors = <&tsens1 4>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpu-1-1-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpuss_1_0_thermal: cpuss-1-0-thermal {
+			thermal-sensors = <&tsens1 5>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpuss-1-0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		cpuss_1_1_thermal: cpuss-1-1-thermal {
+			thermal-sensors = <&tsens1 6>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				cpuss-1-1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		aoss2-thermal {
+			thermal-sensors = <&tsens2 0>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				aoss2-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		gpuss0_thermal: gpuss0-thermal {
+			thermal-sensors = <&tsens2 1>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				gpuss0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		gpuss1_thermal: gpuss1-thermal {
+			thermal-sensors = <&tsens2 2>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				gpuss1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		gpuss2_thermal: gpuss2-thermal {
+			thermal-sensors = <&tsens2 3>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				gpuss2-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		gpuss3_thermal: gpuss3-thermal {
+			thermal-sensors = <&tsens2 4>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				gpuss3-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		gpuss4_thermal: gpuss4-thermal {
+			thermal-sensors = <&tsens2 5>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				gpuss4-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		gpuss5_thermal: gpuss5-thermal {
+			thermal-sensors = <&tsens2 6>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				gpuss5-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		gpuss6_thermal: gpuss6-thermal {
+			thermal-sensors = <&tsens2 7>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				gpuss6-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		gpuss7_thermal: gpuss7-thermal {
+			thermal-sensors = <&tsens2 8>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				gpuss7-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		modem0-thermal {
+			thermal-sensors = <&tsens2 9>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				modem0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		modem1-thermal {
+			thermal-sensors = <&tsens2 10>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				modem1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		modem2-thermal {
+			thermal-sensors = <&tsens2 11>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				modem2-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		modem3-thermal {
+			thermal-sensors = <&tsens2 12>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				modem3-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		camera0-thermal {
+			thermal-sensors = <&tsens2 13>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				camera0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		camera1-thermal {
+			thermal-sensors = <&tsens2 14>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				camera1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		video-thermal {
+			thermal-sensors = <&tsens2 15>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				video-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		aoss3-thermal {
+			thermal-sensors = <&tsens3 0>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				aoss3-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		nsphvx0-thermal {
+			thermal-sensors = <&tsens3 1>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				nsphvx0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		nsphvx1-thermal {
+			thermal-sensors = <&tsens3 2>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				nsphvx1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		nsphvx2-thermal {
+			thermal-sensors = <&tsens3 3>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				nsphvx2-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		nsphmx0-thermal {
+			thermal-sensors = <&tsens3 4>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				nsphmx0-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		nsphmx1-thermal {
+			thermal-sensors = <&tsens3 5>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				nsphmx1-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		nsphmx2-thermal {
+			thermal-sensors = <&tsens3 6>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				nsphmx2-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		nsphmx3-thermal {
+			thermal-sensors = <&tsens3 7>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <5000>; type = "hot"; };
+				nsphmx3-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+		ddr-thermal {
+			thermal-sensors = <&tsens3 8>;
+			trips {
+				trip-point0 { temperature = <120000>; hysteresis = <2000>; type = "hot"; };
+				ddr-critical { temperature = <125000>; hysteresis = <0>; type = "critical"; };
+			};
+		};
+	};
+};
